### PR TITLE
fix: improve transaction logs

### PR DESCRIPTION
**Pull request by UncleBill**
MF-0000

### CODE REVIEW

File: packages/mask/src/extension/popups/pages/Wallet/TransactionDetail/useTransactionLogs.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    const logs = useMemo(() => {
+        if (!transactionState) return EMPTY_LIST
+        let localTransaction: RecentTransaction<ChainId, EvmTransaction> | undefined = undefined
+        const isRecentTx = 'candidates' in transactionState
+        if (isRecentTx) {
+            localTransaction = transactionState
+        }
+        if (!isRecentTx) {
+            localTransaction = txes?.find((x) => x.id === transactionState.hash)
+        }
```
Comment (guanbinrui) [inline packages/mask/src/extension/popups/pages/Wallet/TransactionDetail/useTransactionLogs.ts]:
```ts
const localTransaction = 'candidates' in transactionState ? transactionState : txes?.find((x) => x.id === transactionState.hash)
```

File: packages/mask/src/extension/popups/pages/Wallet/TransactionDetail/useTransactionLogs.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * But only local transaction record activity logs.
+ * Would try to get activity from local record transaction for online transaction
+ */
+export function useTransactionLogs(transactionState: TransactionState) {
+    const { t } = useI18N()
+    const isRecentTx = transactionState && 'candidates' in transactionState
+    const account = useAccount(NetworkPluginID.PLUGIN_EVM)
+    const { Transaction } = useWeb3State(NetworkPluginID.PLUGIN_EVM)
+    const { data: txes } = useQuery({
+        enabled: transactionState && !isRecentTx,
```
Comment (guanbinrui) [inline packages/mask/src/extension/popups/pages/Wallet/TransactionDetail/useTransactionLogs.ts]:
```suggestion
        enabled: transactionState && !('candidates' in transactionState),
```

File: packages/mask/src/extension/popups/pages/Wallet/TransactionDetail/types.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import { type RecentTransaction, type Transaction } from '@masknet/web3-shared-base'
+import { type ChainId, type Transaction as EvmTransaction, type SchemaType } from '@masknet/web3-shared-evm'
+
+export type TransactionState = Transaction<ChainId, SchemaType> | RecentTransaction<ChainId, EvmTransaction> | undefined
```
Comment (guanbinrui) [inline packages/mask/src/extension/popups/pages/Wallet/TransactionDetail/types.ts:4]: Move this type to `packages/mask/src/extension/popups/pages/Wallet/TransactionDetail/useTransactionLogs.ts`

Comment (UncleBill) [reply packages/mask/src/extension/popups/pages/Wallet/TransactionDetail/types.ts:4]: Would be used in `TransactionDetail/index.tsx` as well.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**